app/models/snippet_tag.rb

```diff
@@ -0,0 +1,9 @@
+class SnippetTag < ActiveRecord::Base
+
+  belongs_to :snippet
+  belongs_to :tag
+
+  validates  :snippet_id, presence: true
```

Comment:
It's a good habit to get into to validate the object rather than the id - so validates :snippet rather than :snippet_id - the reason is that validates that the snippet_id value points at a valid snippet as well as being not null.